packages/utils/telemetry-utils/src/sampledTelemetryHelper.ts

```diff
@@ -125,25 +206,78 @@ export class SampledTelemetryHelper implements IDisposable {
 		}
 
 		if (m.count >= this.sampleThreshold) {
+			// Computation extracted separately to avoid multiple division operations.
+			if (this.includeAggregateMetrics) {
+				m.averageDuration = (m.totalDuration ?? 0) / m.count;
+			}
 			this.flushBucket(bucket);
 		}
 
 		return returnValue;
 	}
 
+	private accumulateCustomData(customData: TCustomMetrics, bucket: string): LoggerData {
+		let loggerData = this.measurementsMap.get(bucket);
+		if (loggerData === undefined) {
+			loggerData = {
+				measurements: { count: 0, duration: -1 },
+				dataSums: {},
+				dataMaxes: {},
+			};
+			this.measurementsMap.set(bucket, loggerData);
+		}
+
+		for (const [key, val] of Object.entries(customData)) {
+			assert(typeof key === "string", "Key should be a string");
+			assert(typeof val === "number", "Value should be a number");
+
+			loggerData.dataSums[key] = (loggerData.dataSums[key] ?? 0) + val;
+			loggerData.dataMaxes[key] = Math.max(
+				loggerData.dataMaxes[key] ?? Number.NEGATIVE_INFINITY,
+				val,
+			);
+		}
+
+		return loggerData;
+	}
+
+	private processCustomData(loggerData: LoggerData, count: number): Record<string, number> {
+		const processedCustomData: Record<string, number> = {};
+
+		if (loggerData.dataSums === undefined || loggerData.dataMaxes === undefined) {
+			return processedCustomData;
```

Comment:
I can change this to `return {} satisfies Record<string, number>;`. 

However, I believe the performance improvement will be minimal and thus prefer creating `processedCustomData` with the invocation of the method as I believe this is more readable.